pkgs/common-updater/unstable-updater.nix

```diff
@@ -78,27 +88,51 @@ let
     pushd "$tmpdir"
     commit_date="$(${git}/bin/git show -s --pretty='format:%cs')"
     commit_sha="$(${git}/bin/git show -s --pretty='format:%H')"
-    if [[ -z "$use_stable_version" ]]; then
-        new_version="unstable-$commit_date"
-    else
-        depth=100
-        while (( $depth < 10000 )); do
-            last_tag="$(${git}/bin/git describe --tags --abbrev=0 2> /dev/null || true)"
-            if [[ -n "$last_tag" ]]; then
-                break
+    last_tag=""
+    if [[ -z "$hardcode_zero_version" ]]; then
+        if [[ "$shallow_clone" == "1" ]]; then
+            depth=100
+            while (( $depth < 10000 )); do
+                last_tag="$(${git}/bin/git describe --tags --abbrev=0 2> /dev/null || true)"
+                if [[ -n "$last_tag" ]]; then
+                    tag_matched="$(echo ''${last_tag} | grep -Eo ''${tag_format} || true)"
+                    if [[ -z "$tag_matched" ]]; then
+                        ${lib.getExe git} tag -d ''${last_tag}
+                        continue
+                    fi
+                    break
+                fi
+                ${git}/bin/git fetch --depth="$depth" --tags
+                depth=$(( $depth * 2 ))
+            done
+
+            if [[ -z "$last_tag" ]]; then
+                # To be extra sure, check if full history helps with finding a tag
+                git fetch --tags
+                last_tag="$(${git}/bin/git describe --tags --abbrev=0 2> /dev/null || true)"
```

Comment:
shouldn't this also respect `tag_format`?

Reply:
also, we can simplify the `tag_format` check by passing `--match` to the git command

Reply:
Implemented with `--match` now, and uses `--match` here.